### ISSUE TITLE
TypeError: typing.ClassVar[list[mirascope.core.base.toolkit.ToolKitToolMethod]] is not valid as type argument

### ISSUE THREAD

**Issue by betterthanever2** [Mirascope/mirascope#474]
### Description

Please, help, I started getting this error out of nowhere. Full trace:
```
  File "/usr/local/lib/python3.11/site-packages/scomps/extraction/__init__.py", line 1, in <module>

    from scomps.extraction import core

  File "/usr/local/lib/python3.11/site-packages/scomps/extraction/__init__.py", line 1, in <module>

    from scomps.extraction import core

  File "/usr/local/lib/python3.11/site-packages/scomps/extraction/core.py", line 1, in <module>

    from mirascope.integrations.tenacity import collect_errors

  File "/usr/local/lib/python3.11/site-packages/mirascope/__init__.py", line 7, in <module>

    from . import core as core

  File "/usr/local/lib/python3.11/site-packages/mirascope/core/__init__.py", line 5, in <module>

    from . import base

  File "/usr/local/lib/python3.11/site-packages/mirascope/core/base/__init__.py", line 24, in <module>

    from .toolkit import BaseToolKit, toolkit_tool

  File "/usr/local/lib/python3.11/site-packages/mirascope/core/base/toolkit.py", line 32, in <module>

    class BaseToolKit(BaseModel, ABC):

  File "/usr/local/lib/python3.11/site-packages/pydantic/_internal/_model_construction.py", line 114, in __new__

    private_attributes = inspect_namespace(

                         ^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/site-packages/pydantic/_internal/_model_construction.py", line 448, in inspect_namespace

    ann_type = eval_type_backport(

               ^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/site-packages/pydantic/_internal/_typing_extra.py", line 279, in eval_type_backport

    return _eval_type_backport(value, globalns, localns, type_params)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/site-packages/pydantic/_internal/_typing_extra.py", line 303, in _eval_type_backport

    return _eval_type(value, globalns, localns, type_params)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/site-packages/pydantic/_internal/_typing_extra.py", line 332, in _eval_type

    return typing._eval_type(  # type: ignore

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/typing.py", line 371, in _eval_type

    return t._evaluate(globalns, localns, recursive_guard)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/typing.py", line 876, in _evaluate

    type_ = _type_check(

            ^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/typing.py", line 189, in _type_check

    raise TypeError(f"{arg} is not valid as type argument")

TypeError: typing.ClassVar[list[mirascope.core.base.toolkit.ToolKitToolMethod]] is not valid as type argument

Unable to evaluate type annotation 'ClassVar[list[ToolKitToolMethod]]'.
```

**Comment by betterthanever2** [comment]
Fixing Pydantic version at 2.7.4 helped to get things back on track

**Comment by willbakst** [comment]
Can you please share the code that produces this error (or at least a minimal repro) so I can look into the issue?

**Comment by willbakst** [comment]
Ok I was able to reproduce. This looks like a reported issue in pydantic `v2.9`

See: https://github.com/pydantic/pydantic/issues/10345

**Comment by betterthanever2** [comment]
> Ok I was able to reproduce. This looks like a reported issue in pydantic `v2.9`
> 
> See: [pydantic/pydantic#10345](https://github.com/pydantic/pydantic/issues/10345)

I suspected as much. It was released yesterday, the timing fits

**Comment by willbakst** [comment]
Pydantic released the fix for this in `2.9.1` and my checks have confirmed things seem to be working, so I am closing this issue.
